catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/CatalogWriterCallable.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.migratable.impl;
+
+import java.io.File;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import ddf.catalog.data.Result;
+
+class CatalogWriterCallable implements Callable<Void> {
+    private final File exportFile;
+
+    private final List<Result> fileResults;
+
+    private final MigrationFileWriter fileWriter;
+
+    public CatalogWriterCallable(final File exportFile, final List<Result> fileResults,
+            final MigrationFileWriter fileWriter) {
+        this.exportFile = exportFile;
+        this.fileResults = fileResults;
```

Comment:
Since this is an internal class not meant to be used outside this package, should we just add a comment saying that the class expects the list to stay immutable instead of incurring the cost of making a defensive copy?